api/src/test/scala/api/protocol/HmdaApiProtocolSpec.scala

```diff
@@ -0,0 +1,18 @@
+package hmda.api.protocol
+
+import scala.concurrent.ExecutionContext
+import org.scalatest._
+import prop._
+import hmda.api.model.{ Status, ModelGenerators }
+import spray.json._
+
+class HmdaApiProtocolSpec extends PropSpec with PropertyChecks with MustMatchers with ModelGenerators with HmdaApiProtocol {
+
+  property("status must convert to and from json") {
+    forAll(statusGen) { (s) =>
```

Comment:
Had to read up on [generator-driven property checks](http://www.scalatest.org/user_guide/generator_driven_property_checks) to figure out what's going on here.  I think I get it. :)  However, one thing it did mention was:

>Note that even if you are use generators that don't produce the invalid values, you still need the `whenever` clause. The reason is that once a property fails, ScalaCheck will try and shrink the values to the smallest values that still cause the property to fail. During this shrinking process ScalaCheck may pass invalid values. The `whenever` clause is still needed to guard against those values. (The `whenever` clause also clarifies to readers of the code exactly what the property is in a succinct way, without requiring that they find and understand the generator definitions.)

Do this apply here?  It's pretty much impossible for this thing to fail at this point, but maybe it's still a _best practice_?